fs-repo-14-to-15/migration/migration.go

```diff
@@ -172,65 +172,79 @@ func convert(in io.Reader, out io.Writer) error {
 	}
 
 	// Remove /quic only addresses from .Addresses.Swarm
-	if a, ok := confMap["Addresses"]; ok {
-		addresses, ok := a.(map[string]any)
-		if !ok {
-			return fmt.Errorf("invalid type for .Addresses got %T expected json map", a)
-		}
-
-		for _, addressToRemove := range [...]string{"Swarm", "Announce", "AppendAnnounce", "NoAnnounce"} {
-			s, ok := addresses[addressToRemove]
+	if err := func() error {
+		if a, ok := confMap["Addresses"]; ok {
+			addresses, ok := a.(map[string]any)
 			if !ok {
-				continue
+				fmt.Printf("invalid type for .Addresses got %T expected json map; skipping .Addresses\n", a)
+				return nil
 			}
 
-			swarm, ok := s.([]interface{})
-			if !ok {
-				return fmt.Errorf("invalid type for .Addresses.%s got %T expected json array", addressToRemove, s)
-			}
+			for _, addressToRemove := range [...]string{"Swarm", "Announce", "AppendAnnounce", "NoAnnounce"} {
+				s, ok := addresses[addressToRemove]
+				if !ok {
+					continue
+				}
+
+				swarm, ok := s.([]interface{})
+				if !ok {
+					fmt.Printf("invalid type for .Addresses.%s got %T expected json array; skipping .Addresses.%s\n", addressToRemove, s, addressToRemove)
+					continue
+				}
 
-			var newSwarm []interface{}
-			for _, v := range swarm {
-				if addr, ok := v.(string); ok {
-					if !strings.Contains(addr, "/quic") || strings.Contains(addr, "/quic-v1") {
-						newSwarm = append(newSwarm, addr)
+				var newSwarm []interface{}
+				for _, v := range swarm {
+					if addr, ok := v.(string); ok {
+						if !strings.Contains(addr, "/quic") || strings.Contains(addr, "/quic-v1") {
```

Comment:
Not sure if we have good options, but it's pretty awkward that we're removing `/quic` here  when we have no idea if there's a substitute `/quic-v1` in place. Should we look for identical matches instead, otherwise we're going to delete the users' data that they may have added to `AppendAnnounce`, etc.

Given the user has to do something here would we be better off:
1. Changing their custom config to `/quic-v1`
2. Deleting `/quic`
3. Letting the user get an error on start and figure it out (error messages in kubo need to be helpful enough for this to be viable, they might already be)

Seems like probably not a ton of people will be impacted, but the support time here might be no fun unless this is taken care of 😢.

Reply:
we already had a migration that duplicated quic into quic-v1 addresses, do you want me to copy paste it and rerun it again ?

Reply:
Well don't want to end up with duplicate quic-v1 addresses. Could do more work here to make sure we're not, but a bit concerned about added surface needing testing.

Might be fine leaving things alone given it meant that after quic-v1 addresses were in the config the user added only a `/quic` address.

WDYT?

Reply:
+1, it will be safer is we effectively re-run migration and convert all `/quic` to `/quic-v1` and then remove any `/quic-1` duplicates. Way safer than blindly removing `/quic` from user config.